server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java

```diff
@@ -92,26 +103,78 @@ public void downloads(String username, String attachmentId) throws Throwable {
-        response = with.get("/download/" + blobId);
+        response = with
+                .get("/download/" + blobId);
+    }
+
+    private String attachmentAccessToken(String blobId, AccessToken accessToken) {
+        return
+            with()
+                .header("Authorization", accessToken.serialize())
+            .post("/download/" + blobId)
+            .then()
+                .extract()
+                .asString();
+    }
+
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with a valid authentication token$")
+    public void downloadsWithValidToken(String username, String unknownBlobId) throws Throwable {
+        String blobId = blobIdByAttachmentId.getOrDefault("2", null);
+        AccessToken accessToken = userStepdefs.tokenByUser.get(username);
+        response = with()
+                .param("access_token", attachmentAccessToken(blobId, accessToken))
+                .get("/download/" + unknownBlobId);
+    }
+
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" without the authentication token$")
+    public void getDownloadWithoutToken(String username, String attachmentId) {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
+        response = with().get("/download/" + blobId);
+    }
+
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with empty authentication token$")
+    public void getDownloadWithEmptyToken(String username, String attachmentId) {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
+        response = with()
+                .param("access_token", "")
+                .get("/download/" + blobId);
+    }
+
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with a bad authentication token$")
+    public void getDownloadWithBadToken(String username, String attachmentId) {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
+        response = with()
+                .param("access_token", "bad")
+                .get("/download/" + blobId);
+    }
+
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with an invalid authentication token$")
+    public void getDownloadWithUnknownToken(String username, String attachmentId) {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
+        response = with()
+                .param("access_token", INVALID_ATTACHMENT_TOKEN)
+                .get("/download/" + blobId);
     }
     
 
-    @When("^\"([^\"]*)\" asks for an attachment without blobId parameter$")
-    public void getDownloadWithoutBlobId(String username) throws Throwable {
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" without blobId parameter$")
+    public void getDownloadWithoutBlobId(String username, String attachmentId) throws Throwable {
+        String blobId = blobIdByAttachmentId.get(attachmentId);
         AccessToken accessToken = userStepdefs.tokenByUser.get(username);
         response = with()
-            .header("Authorization", accessToken.serialize())
+            .param("access_token", attachmentAccessToken(blobId, accessToken))
```

Comment:
Did you really replace actual tests with Authorization headers with access_token param?
I think we should still support Authorization headers